OpenSim/Simulation/Wrap/WrapSphere.cpp

```diff
@@ -589,3 +591,23 @@ int WrapSphere::wrapLine(const SimTK::State& s, SimTK::Vec3& aPoint1, SimTK::Vec
 
    return return_code;
 }
+
+// Implement generateDecorations by WrapSphere to replace the previous out of place implementation 
+// in ModelVisualizer
+void WrapSphere::generateDecorations(bool fixed, const ModelDisplayHints& hints, const SimTK::State& state,
+    SimTK::Array_<SimTK::DecorativeGeometry>& appendToThis) const 
+{
+    if (hints.get_show_wrap_geometry()) {
+        const Vec3 color(SimTK::Cyan);
+        SimTK::Transform ztoy;
+        ztoy.updR().setRotationFromAngleAboutX(SimTK_PI / 2);
+        const SimTK::Transform& X_GB = getFrame().getMobilizedBody().getBodyTransform(state);
```

Comment:
Should also consider having the base Wrap object providing a helper to place the decorative geometry since the code is identical for three type implemented so far. All that differs is the creation of the Decorative shape and seems like that's all the derived classes should specify. I imagine Torus would then follow the same pattern